### DISCUSSION TITLE
How to use inline edit along with renderRowActions.

### DISCUSSION THREAD

**Discussion by sagrawal9689** [KevinVandy/material-react-table#275] [Q&A] [answered]
Currently when I pass the prop enableEditing as true and use the renderRowActions props to add some more buttons to the actions column the editing icon gets overridden by renderRowActions react component. Is there a way I can use the existing editing functionality and add my own custom actions too to the action buttons.

**Comment by KevinVandy** [comment, marked as answer]
`renderRowActions` is kind of an override for that column. You can either implement your own edit button along side the other buttons, or put your buttons in their own display column instead of using renderRowActions.

https://www.material-react-table.com/docs/guides/display-columns#create-your-own-display-columns

**Reply by KevinVandy** [reply]
Though I'm considering adding an export for the editing components in a new release so that you could do something like

```js
renderRowActions={({ row, table }) => <Box><MRT_EditButtons row={row} table={table} /> {YourCustomEditButtons} </Box>}
```

**Reply by sagrawal9689** [reply]
@KevinVandy yes adding the export would be a good idea. Even I was searching if there was any such export for the edit button at first.

**Comment by dmitrymaslov99** [comment]
Hello. I want to use editing row - https://www.material-react-table.com/docs/guides/editing#row-editing-mode
But i have custom actions buttons - https://www.material-react-table.com/docs/guides/row-actions
And it turns out that if I have my own buttons, then editing will not work (there will be no buttons to save or cancel). I threw a small example, what can you advise?
https://codesandbox.io/p/sandbox/busy-carson-dn5zwt?file=%2Fsrc%2FTS.tsx&selection=%5B%7B%22endColumn%22%3A20%2C%22endLineNumber%22%3A55%2C%22startColumn%22%3A20%2C%22startLineNumber%22%3A55%7D%5D
